.github/workflows/test-python-app.yaml

```diff
@@ -0,0 +1,61 @@
+name: Test Python (Django) application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test_django:
+
+    runs-on: ubuntu-latest
+    
+    # setup the postgres service needed by the webapp
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: mp
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        
+    # setup env for the application
+    env:
+      DJANGO_SETTINGS_MODULE: 'settings'
+      DB_HOST: 127.0.0.1
+      DB_PASSWORD: postgres
+      DB_NAME: mp
+      DB_USER: postgres
+      SECRET_KEY: thISkEyisUsedForTravis
+      SOLR_URL: http://127.0.0.1:8983/solr/tester
+      SOLR_ADMIN_URL: http://127.0.0.1:8983/solr/admin/cores
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        cd app
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install codecov
+    - name: Create and run migrations
+      run: |
+        cd app
+        python manage.py makemigrations --noinput
+        python manage.py migrate --noinput
+    - name: Run collectstatic
+      run: python app/manage.py collectstatic --noinput
+    - name: Run tests
+      run: |
+        cd app
+        coverage run --source=. manage.py test
+        coverage report
```

Comment:
Do we also need this? (I don't know, I haven't used it before)
* https://github.com/codecov/codecov-action#usage

Reply:
> Do we also need this? (I don't know, I haven't used it before)

I've used it a few times in the past with pytest ([example](https://github.com/harens/checkdigit/blob/master/.github/workflows/test.yml)).

I wouldn't use `coverage report`, since at least to my knowledge, this only outputs the results in the CI logs and doesn't upload it.

Maybe do something like the following to generate a coverage xml file.

```sh
cd app
# For --branch, see https://coverage.readthedocs.io/en/coverage-5.5/branch.html#branch
coverage run --branch --source=. manage.py test
# Adjust fail under number to your liking (currently 100% coverage required)
coverage xml --fail-under=100
```

This file can then be uploaded to codecov. This has the added benefit that no credentials are required, and the codecov bot will then work on PRs:

```yaml
- name: Upload coverage to Codecov
        uses: codecov/codecov-action@v1.5.2
        with:
          file: ./coverage.xml
          flags: unittests
          fail_ci_if_error: true
```

Reply:
Oh, thank you! I have updated the PR with uploads to codecov, but reporting will take place after there is a base to compare with. Seems like we can merge it now.

Reply:
Sure. We can still make further improvements later.